### PR TITLE
[proxy] Enable Caddy metrics

### PR DESCRIPTION
**Pull request by corneliusludmann**
## Description

This PR enables the metrics endpoint on port `8003` in order to be consumed in Prometheus.

~~@ArthurSens Could you test it with the preview environment?~~ Done.

@aledbf Could you review this?

## Related Issue(s)
See also:
- https://github.com/gitpod-io/ops/pull/9
- https://github.com/gitpod-io/observability/issues/265

## Release Notes

```release-note
NONE
```

/no-cc

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **corneliusludmann** after the PR has been reviewed.

Associated issue: *#265*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/proxy/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/proxy/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by ArthurSens — APPROVED** [review]
From my side, it looks good! Metrics are being exposed.

We'll still need approval from the `OWNERS`

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 540673f2fd5e769477d90f26a7bccf1855f2dd9e</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5739#pullrequestreview-757412070" title="Approved">ArthurSens</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **corneliusludmann** after the PR has been reviewed.

Associated issue: *#265*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/proxy/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/proxy/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by aledbf** [comment]
@corneliusludmann I found the default metrics too verbose and a bit useless (`server="srv1"`) 
That's the reason why I didn't enable the metrics during the migration from nginx https://caddy.community/t/prometheus-server-field-value/12171

**Comment by ArthurSens** [comment]
> @corneliusludmann I found the default metrics too verbose and a bit useless (`server="srv1"`)
> That's the reason why I didn't enable the metrics during the migration from nginx https://caddy.community/t/prometheus-server-field-value/12171

In case we're really worried about caddy metrics being too expensive, I can configure Prometheus to drop the expensive metrics.
But we really need those metrics at the moment because we're kinda clueless about the number of go_routines that caddy will start using now that we're caching OpenVSX requests.

**Comment by aledbf** [comment]
> In case we're really worried about caddy metrics being too expensive, I can configure Prometheus to drop the expensive metrics.

We will pay the price of generating the metrics in the caddy pod.

**Comment by ArthurSens** [comment]
Do you mean that scraping metrics will have an impact on the performance of caddy?

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5739#issuecomment-921927017" title="Approved">aledbf</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/5739#pullrequestreview-757412070" title="Approved">ArthurSens</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **corneliusludmann** after the PR has been reviewed.

Associated issue: *#265*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/proxy/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/proxy/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by aledbf** [comment]
We cannot disable the default Caddy metrics. For that reason, let's get this in to start gathering details about the go routines.

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5739#issuecomment-921927017" title="Approved">aledbf</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/5739#pullrequestreview-757412070" title="Approved">ArthurSens</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/5739#issuecomment-922060843" title="Approved">corneliusludmann</a>*

Associated issue: *#265*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/proxy/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/proxy/OWNERS)~~ [corneliusludmann]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
